### PR TITLE
nixos/tests/jellyfin: enhanced test

### PR DESCRIPTION
**Pull request by minijackson**
###### Motivation for this change

Discovered the Jellyfin API today, and thought it would be nice to have a test that tests more than only an HTTP request.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @nyanloutre, @purcell

### CODE REVIEW

**Review by purcell — APPROVED** [review]
Good thinking, PR looks good to me!

**Comment by lukegb** [comment]
@ofborg test jellyfin

File: nixos/tests/jellyfin.nix [hunk 1]
```diff
@@ -1,16 +1,151 @@
...
+        import time
+        from urllib.parse import urlencode
+
+        machine.wait_for_unit("jellyfin.service")
+        machine.wait_for_open_port(8096)
+        machine.succeed("curl --fail http://localhost:8096/")
+
+        time.sleep(1)
+
+        health = machine.succeed("curl --fail http://localhost:8096/health")
```
Comment (lukegb) [inline nixos/tests/jellyfin.nix:46]: Maybe

```suggestion
        machine.wait_until_succeeds("curl --fail http://localhost:8096/health | grep Healthy")
```

instead - and drop the time.sleep above.

File: nixos/tests/jellyfin.nix [hunk 2]
```diff
@@ -1,16 +1,151 @@
...
+            )
+
+            machine.succeed(
+                f"${postWith
+                  "/Library/VirtualFolders?{add_folder_query}"
+                  payloads.empty}"
+            )
+
+        # Wait for library refresh
+        time.sleep(1)
```
Comment (lukegb) [inline nixos/tests/jellyfin.nix:111]: Is there an API you can use to detect when it's finished?

**Review by lukegb — CHANGES_REQUESTED** [review]
My main concerns are the use of sleeps; on oversubscribed machines 1s may not be enough which will cause the test to fail (and it's a bit brittle anyway)

**Comment by minijackson** [comment]
Thanks for the great comments @lukegb, they definitely made the test better

**Review by lukegb — APPROVED** [review]

**Comment by lukegb** [comment]
@ofborg test jellyfin

**Pull request merged**